broker/src/main/java/org/apache/rocketmq/broker/failover/EscapeBridge.java

```diff
@@ -112,15 +113,21 @@ public PutMessageResult putMessage(MessageExtBrokerInner messageExt) {
     }
 
     private SendResult putMessageToRemoteBroker(MessageExtBrokerInner messageExt) {
-        final TopicPublishInfo topicPublishInfo = this.brokerController.getTopicRouteInfoManager().tryToFindTopicPublishInfo(messageExt.getTopic());
+        final boolean isTransHalfMessage = TransactionalMessageUtil.buildHalfTopic().equals(messageExt.getTopic());
+        MessageExtBrokerInner messageToPut = messageExt;
+        if (isTransHalfMessage) {
+            messageToPut = TransactionalMessageUtil.buildTransactionalMessageFromHalfMessage(messageExt);
+        }
```

Comment:
How about moving this logic to the upper layer to make putMessageToRemoteBroker function purer？